server.go

```diff
@@ -324,190 +324,64 @@ func (s *Server) handlerAddMovie(w http.ResponseWriter, r *http.Request) {
-			} else if len(data.ValTitle) == 0 {
-				s.l.Debug("Title too short: %d", len(common.CleanMovieName(data.ValTitle)))
-				data.ErrTitle = true
-				errText = append(errText, "Title too short!")
-			}
+				// Prepare a int for the id
+				var movieId int
 
-			movieExists, err := s.data.CheckMovieExists(r.FormValue("MovieName"))
-			if err != nil {
-				s.doError(
-					http.StatusInternalServerError,
-					fmt.Sprintf("Unable to check if movie exists: %v", err),
-					w, r)
-				return
-			}
-
-			if movieExists {
-				data.ErrTitle = true
-				s.l.Debug("Movie exists")
-				errText = append(errText, "Movie already exists")
-			}
-
-			if data.ValTitle == "" && !(r.FormValue("AutofillBox") == "on") {
-				errText = append(errText, "Missing movie title")
-				data.ErrTitle = true
-			}
-
-			descr := strings.TrimSpace(r.FormValue("Description"))
-			data.ValDescription = descr
-
-			maxDescriptionLength, err := s.data.GetCfgInt(ConfigMaxDescriptionLength, DefaultMaxDescriptionLength)
-			if err != nil {
-				s.l.Error("Unable to get %q: %v", ConfigMaxTitleLength, err)
-				s.doError(
-					http.StatusInternalServerError,
-					"something went wrong :C",
-					w, r)
-				return
-			}
-
-			if len(data.ValDescription) > maxDescriptionLength {
-				s.l.Debug("Description too long: %d", len(data.ValDescription))
-				data.ErrDescription = true
-				errText = append(errText, "Description too long!")
-			}
+				movieId, err = s.data.AddMovie(movie)
+				if err != nil {
+					s.l.Error("Movie could not be added. Error: %v", err)
+				} else {
+					http.Redirect(w, r, fmt.Sprintf("/movie/%d", movieId), http.StatusFound)
+				}
 
-			if len(descr) == 0 && !(r.FormValue("AutofillBox") == "on") {
-				data.ErrDescription = true
-				errText = append(errText, "Missing description")
 			}
+		} else if formfillEnabled {
+			s.l.Debug("formfill")
+			// do formfill
+			results, links := s.handleFormfill(&data, w, r)
 
-			movie.Name = strings.TrimSpace(r.FormValue("MovieName"))
-			movie.Description = strings.TrimSpace(r.FormValue("Description"))
-			movie.Links = links
-			movie.Remarks = remarks
-
-			posterFileName := strings.TrimSpace(r.FormValue("MovieName"))
-
-			posterFile, _, _ := r.FormFile("PosterFile")
+			if results == nil || links == nil {
+				data.ErrorMessage = append(data.ErrorMessage, "One or more fields reported an error.")
+			} else {
+				// Fill all the fields in the movie struct
+				movie.Name = results[0]
+				movie.Description = results[1]
+				movie.Poster = filepath.Base(results[2])
+				movie.Remarks = results[3]
+				movie.Links = links
+				movie.AddedBy = user
 
-			if posterFile != nil {
-				file, err := s.uploadFile(r, posterFileName)
+				// Prepare a int for the id
+				var movieId int
 
+				movieId, err = s.data.AddMovie(movie)
 				if err != nil {
-					data.ErrPoster = true
-					errText = append(errText, err.Error())
+					s.l.Error("Movie could not be added. Error: %v", err)
```

Comment:
Same as above error (show something generic to the user).

Reply:
Will be changed in the next commit